### ISSUE TITLE
restrict-plus-operands does not report explicit any when allowAny is false

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1228]
## Summary

Reproduced with @utoo/lint 0.4.1 on Node.js 22.23.1, macOS arm64. The operand is explicitly annotated any. With allowAny: false it should be rejected. Using an explicit option avoids a comparison based on different plugin-version defaults.

## Minimal reproduction

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0
```

`example.tsx`:

```tsx
export function example(value: any) { return value + 1; }
```

`tsconfig.json`:

```json
{"compilerOptions":{"target":"ES2022","strict":true},"include":["example.tsx"]}
```

`utlint.config.json`:

```json
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    },
    "project": "./tsconfig.json"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
};
```

```sh
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
```

## Result

ESLint reports one @typescript-eslint/restrict-plus-operands error. utlint reports no diagnostics.

Both tools check the same file successfully; there are no syntax errors or lint-disable comments.

**Comment by fireairforce** [comment]
Retested with @utoo/lint 0.4.2 using the original issue setup and public reference package versions. The explicitly annotated any operand in the original reproduction is fixed. Implicit any parameters and member access through an as-any assertion still receive no diagnostic with allowAny:false.

Run the following examples separately as example.tsx.

### plus-implicit

```tsx
export function example(value){return value+1;}
```

Rule configuration for both tools:

```json
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
}
```

ESLint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.
```

utlint:

```text
(no diagnostics)
```

### plus-assertion

```tsx
export function example(value:unknown){return (value as any).count+1;}
```

Rule configuration for both tools:

```json
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
}
```

ESLint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.
```

utlint:

```text
(no diagnostics)
```

The reference uses @typescript-eslint/parser and eslint-plugin 6.21.0 with parserOptions.project pointing to a tsconfig.json that includes example.tsx and enables strict:true. Both snippets parse successfully. Explicit allowAny:false avoids plugin-version default differences.
